update_accounts.py

```diff
@@ -30,42 +30,45 @@ def refresh_token(api_url: str, refresh_token: str, client_id: str, client_secre
-        TOKEN = json.load(token_file)
-    with open(args.oauth_settings) as oauth_file:
-        OAUTH = json.load(oauth_file)
-
-    access_token = TOKEN.get("access_token")
+def update_discovery_url(api_url: str, discovery_url: str, token: dict, settings: dict):
+    log.info(f"Updating discovery url on {api_url} to '{discovery_url}'")
+    access_token = token.get("access_token")
     try:
-        CANVAS = Canvas(api_url, access_token)
-        account = CANVAS.get_account(1)
+        canvas = Canvas(api_url, access_token)
+        account = canvas.get_account(1)
     except InvalidAccessToken:
         log.info("Access token is invalid, attempting to refresh!")
-        access_token = refresh_token(api_url, TOKEN.get("refresh_token"),
-                                     OAUTH.get("client_id"), OAUTH.get("client_secret"), OAUTH.get("redirect_uri"))
-        CANVAS = Canvas(api_url, access_token)
-        account = CANVAS.get_account(1)
+        access_token = refresh_token(api_url, token.get("refresh_token"),
+                                     settings.get("client_id"), settings.get("client_secret"), settings.get("redirect_uri"))
+        log.debug(access_token)
+        canvas = Canvas(api_url, access_token)
+        account = canvas.get_account(1)
 
+    try:
         # TODO: Make this URL configurable
         account.update_account_auth_settings(
             sso_settings={"auth_discovery_url": discovery_url})
+    except Exception as e:
+        log.error(f"Error updating {api_url} {e.message}")
 
-parser = argparse.ArgumentParser(description='Update Canvas Test/Beta Settings')
-parser.add_argument('canvas_token', type=str, help='Canvas Token File')
-parser.add_argument('oauth_settings', type=str, help='OAuth Settings File')
+def main():
+    parser = argparse.ArgumentParser(description='Update Canvas Test/Beta Settings')
+    parser.add_argument('canvas_token', type=str, help='Canvas Token File')
+    parser.add_argument('settings', type=str, help='Settings File')
 
-args = parser.parse_args()
+    args = parser.parse_args()
 
-if not os.path.isfile(args.canvas_token):
-    log.warn("Canvas Token file does not exist, must provide the path to this file")
-    sys.exit()
-if not os.path.isfile(args.oauth_settings):
-    log.warn("OAuth Settings file does not exist, must provide the path to this file")
-    sys.exit()
-
-# TODO: Make this URL configurable?
-update_discovery_url("https://umich.beta.instructure.com", "")
-update_discovery_url("https://canvas-test.it.umich.edu", "https://canvas-dev.dsc.umich.edu/")
+    try:
+        with open(args.canvas_token) as token_file:
+            token = json.load(token_file)
+        with open(args.settings) as settings_file:
+            settings = json.load(settings_file)
+    except IOError:
+        log.exception("Could not read one of the files provided as arguments.")
+        sys.exit()
 
+    for canvas_server, discovery_url  in settings.get("update_discovery_urls", {}).items():
```

Comment:
Can we add some log messages here, in case "update_discovery_urls" value is empty or missing in the oauth_settings.json file?